firrtl/src/test/scala/firrtlTests/SerializerSpec.scala

```diff
@@ -82,6 +82,46 @@ object SerializerSpec {
       "test"
     )
 
+  val probeModule: String =
+    """module probeModule :
+      |  input clock : Clock
+      |  input cond : UInt<1>
+      |  input in : UInt<8>
+      |  output out : UInt<8>
+      |  output outProbe : RWProbe<UInt<8>>
+
+      |  inst c of child
+      |  define c.in = probe(in)
+      |  out <= read(c.out)
+      |  force_initial(outProbe, UInt<8>("h64"))
+      |  release_initial(outProbe)
+      |  force(clock, cond, outProbe, in)
+      |  release(clock, cond, outProbe)""".stripMargin
+
+  val probeModuleIR: Module =
+    Module(
+      NoInfo,
+      "probeModule",
+      Seq(
+        Port(NoInfo, "clock", Input, ClockType),
+        Port(NoInfo, "cond", Input, UIntType(IntWidth(1))),
+        Port(NoInfo, "in", Input, UIntType(IntWidth(8))),
+        Port(NoInfo, "out", Output, UIntType(IntWidth(8))),
+        Port(NoInfo, "outProbe", Output, RWProbeType(UIntType(IntWidth(8))))
+      ),
+      Block(
+        Seq(
+          DefInstance("c", "child"),
+          ProbeDefine(NoInfo, SubField(Reference("c"), "in"), ProbeExpr(Reference("in"))),
+          Connect(NoInfo, Reference("out"), ProbeRead(Reference("c.out"))),
+          ProbeForceInitial(NoInfo, Reference("outProbe"), UIntLiteral(100, IntWidth(8))),
+          ProbeReleaseInitial(NoInfo, Reference("outProbe")),
+          ProbeForce(NoInfo, Reference("clock"), Reference("cond"), Reference("outProbe"), Reference("in")),
+          ProbeRelease(NoInfo, Reference("clock"), Reference("cond"), Reference("outProbe"))
+        )
+      )
+    )
+
```

Comment:
I'm not against this larger test per se, but at the true unit test level, it's not clear to me that this needs to be such a large example, you could just do stuff like:

```scala
ProbeDefine(NoInfo, SubField(Reference("c"), "in"), ProbeExpr(Reference("in"))).serialize should be("define c.in = probe(in)")
```
Something like this for each of the specific IR nodes you're adding would be sufficient IMO